crates/evm/fuzz/src/strategies/calldata.rs

```diff
@@ -1,18 +1,65 @@
-use super::fuzz_param;
+use crate::strategies::{fuzz_param, EvmFuzzState};
 use alloy_dyn_abi::JsonAbiExt;
 use alloy_json_abi::Function;
-use alloy_primitives::Bytes;
+use alloy_primitives::{Address, Bytes};
+use foundry_config::FuzzDictionaryConfig;
+use hashbrown::HashSet;
 use proptest::prelude::{BoxedStrategy, Strategy};
+use std::{fmt, sync::Arc};
+
+pub type CalldataFuzzDictionary = Arc<CalldataFuzzDictionaryConfig>;
+
+#[derive(Clone)]
+pub struct CalldataFuzzDictionaryConfig {
+    /// Addresses that can be used for fuzzing calldata.
+    pub addresses: Vec<Address>,
+}
+
+impl fmt::Debug for CalldataFuzzDictionaryConfig {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("CalldataFuzzDictionaryConfig").field("addresses", &self.addresses).finish()
+    }
+}
+
+impl CalldataFuzzDictionaryConfig {
+    pub fn new(config: &FuzzDictionaryConfig, state: EvmFuzzState) -> Self {
```

Comment:
please add some docs

Reply:
added in https://github.com/foundry-rs/foundry/pull/7240/commits/b77dcb6fca13e732067cf50ac517e6c5fc641122